### DISCUSSION TITLE
how do i return a query as a array of objects with a plpgsql function

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#5446] [Questions]
when i run this code it returns a table with records as exptected
`select trips.*, vessel.vessel_name from trips inner join vessel ON vessel.id = trips.vessel_id where depature_date_time::timestamp::date='2022-02-13T00:00:00.000Z' AND arrival_place='AA. Thoddoo' AND depature_place='K. Male'`

but now i need to make a plpgsql function to return this table as a array of objects when i call .rpc function...

**Comment by GaryAustin1** [comment]
You would need to use a table type return definition..
RETURNS TABLE (col1 integer
               , col2 integer
               , col3 varchar
               , col4 varchar    ) AS
I don't think there is an option get all of your "trips.*" columns automatically, so you have to list them one at a time, then add your extra column.  The order is all that counts, not the names themselves.
https://dba.stackexchange.com/questions/155955/postgres-function-return-table-with-extra-column
